### ISSUE TITLE
UI breaks on very wide screen

### ISSUE THREAD

**Issue by carlpett** [iverberk/nomad-ui#91]
I'm blessed with a 3440x1440 screen at work. Looking at the ui at this width causes overflowing in the Job overview:

![image](https://cloud.githubusercontent.com/assets/214867/18922437/eba06070-85a8-11e6-9330-7d72335dfef6.png)

I tried figuring out what causes this, but so far I'm stumped. The charts, while too large for their containers, do not cause all the extra height to be added as far as I can tell.

**Comment by iverberk** [comment]
That's a big screen! :-D

I might have trouble reproducing this since I can't replicate the resolution but I agree that the charts are a little bit funky. Are you at all able to investigate this yourself? Maybe by tweaking the css a little bit in the browser. Let me know and I'll see what I can think of.

**Comment by carlpett** [comment]
I'm doing some investigations now, I'll see what I can find. There was nothing obvious from using the dev console, but I'm taking a deeper look into the code instead.

**Comment by carlpett** [comment]
I've found the issue, but spent several hours over many days trying to actually get my fix to run. It seems the webpack-stuff somehow magically reverts changes to dashboard.css? Also, the go-bindata stuff doesn't work on Windows, apparently. Maybe an app run on a local network doesn't need to have all those bells and whistles in order to make it simpler?
I'll try to run it on Linux instead, but the getting started to contribute has not been without its pains.

**Comment by jippi** [comment]
I know what the problem is :) I'll try to fix it this week. :)

**Comment by jippi** [comment]
@carlpett in regards to getting up and running, clone the repo and do the following 

- open /Makefile and comment the 3 version checks (make sure you aprox these versions though :)) 
- `make build`
- `cd frontend ; npm start`
- (start a new tab and nagivate to the root project)
- `cd backend ; NOMAD_ADDR=http://<your nomad cluster ip>:4646 ./build/nomad-ui-<your arch>` 

Thats what i've been doing so far.. i think the project heavily assume you run linux or osx though :)

**Comment by carlpett** [comment]
Ok, so I just can't get this to work properly in my Windows environment, and I don't have a Linux machine available right now (sigh). (I tried with the new Bash on Windows 10 thing, but it does not fully implement some network stuff, and apparently there is a dependency called `macaddress.js` which does not compile if there is no network interface...).

So, the issue is with the selectors ` .ct-perfect-fourth:before` and `.card .ct-chart`. Removing `padding-bottom: 75%`  and `height: 160px` respectively fixes the issue.

**Comment by jippi** [comment]
@carlpett fyi, if you use `latest` in ~5min this should be fixed :)

**Comment by carlpett** [comment]
I'll take you up on that! :)

**Comment by jippi** [comment]
@carlpett there you go   https://hub.docker.com/r/iverberk/nomad-ui/tags/

**Comment by carlpett** [comment]
@jippi Hm, the layout is at least completely different, but it does not seem to actually work? Everything is blank. 
Console output: `TypeError: Cannot read property 'Summary' of null`. I can sanitize the stack trace and upload it if it is useful, but the minified stuff doesn't seem to say much. The container doesn't have any output besides "Connecting to nomad" and "Listening..."

**Comment by carlpett** [comment]
Should note though that it does list my jobs, so it can connect to Nomad. Just something with the overview that is broken.

**Comment by jippi** [comment]
@carlpett what nomad version do you use?

**Comment by carlpett** [comment]
`0.4.0`

**Comment by jippi** [comment]
currently we depend on `0.4.1` which provide some job summaries we use for statistics

**Comment by jippi** [comment]
I can try to hack around the issue in a few

**Comment by carlpett** [comment]
Aha. Ok, I'm not going to upgrade the cluster at 7:30 a friday night, though :)

**Comment by jippi** [comment]
@carlpett I'm going to merge https://github.com/iverberk/nomad-ui/pull/113 as a fix once Jenkins is done :)

**Comment by jippi** [comment]
@carlpett once the merge commit is finished, try and pull `latest` again :)

**Comment by jippi** [comment]
@carlpett try `latest` now :)

**Comment by carlpett** [comment]
Sorry, still doesn't work. Actually seems worse - I can no longer visit the other tabs. Clicking the links in the sidebar gives `Cannot read property 'getHostNode' of null`. Clicking "Cluster" again gives a minified error with a link to the react homepage and the de-minified error `Attempted to update component `Connect(t` that has already been unmounted (or failed to mount).` (probably just a following error from the first one). 

Btw, one thing that might be helpful would be to include the git hash somewhere so that I can verify what version I am on. When restarting a few times now, occasionally I got something cached from a previous version (which may be something to track in a different issue)

**Comment by jippi** [comment]
@carlpett what browser do you use?

**Comment by jippi** [comment]
@carlpett i can reproduce it, will push a fix now

**Comment by jippi** [comment]
@carlpett found a fix, just pushed it to https://github.com/iverberk/nomad-ui/pull/115  should be in dockerhub in ~10min

**Comment by jippi** [comment]
@carlpett its now live on dockerhub :)

**Comment by carlpett** [comment]
@jippi Now it works! Great :)
I normally use Chrome, but when I had the cache issues I tested IE. Surprising noone, it doesn't work. While it doesn't feel like a prime target (esp if the project can't even be developed on Windows), it is a bit strange that it doesn't work. Just white page, no errors in console. I'll file a separate issue.
